File: Maximum Matching of Players With Trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort()
        trainers.sort()
        pidx, tidx = 0, 0
        count = 0
        n = len(players)
        m = len(trainers)
        while pidx < n:
            while tidx < m and trainers[tidx] < players[pidx]:
                tidx += 1

            if tidx < m and players[pidx] <= trainers[tidx]:
                tidx += 1
                count += 1
            pidx += 1
        return count

File: Maximum Matching of Players With Trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        sort(players.begin(), players.end());
        sort(trainers.begin(), trainers.end());

        size_t n = players.size(), m = trainers.size();
        size_t pidx = 0, tidx = 0;
        int count = 0;
        while (pidx < n) {
            while (tidx < m && players[pidx] > trainers[tidx]) tidx++;

            if (tidx < m && players[pidx] <= trainers[tidx]) {
                tidx++;
                count++;
            }
            pidx++;
        }
        return count;
    }
};
